### PR TITLE
[MINOR] Spelling bin core docs external mllib repl

### PR DESCRIPTION
**Pull request by jsoref**
### What changes were proposed in this pull request?

This PR intends to fix typos in the sub-modules:
* `bin`
* `core`
* `docs`
* `external`
* `mllib`
* `repl`
* `pom.xml`

Split per @srowen https://github.com/apache/spark/pull/30323#issuecomment-728981618

NOTE: The misspellings have been reported at https://github.com/jsoref/spark/commit/706a726f87a0bbf5e31467fae9015218773db85b#commitcomment-44064356

### Why are the changes needed?

Misspelled words make it harder to read / understand content.

### Does this PR introduce _any_ user-facing change?

There are various fixes to documentation, etc...

### How was this patch tested?

No testing was performed

### CODE REVIEW

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -333,7 +333,7 @@ license: |
         </tr>
     </table>
 
-  - Since Spark 2.4, when there is a struct field in front of the IN operator before a subquery, the inner query must contain a struct field as well. In previous versions, instead, the fields of the struct were compared to the output of the inner query. Eg. if `a` is a `struct(a string, b int)`, in Spark 2.4 `a in (select (1 as a, 'a' as b) from range(1))` is a valid query, while `a in (select 1, 'a' from range(1))` is not. In previous version it was the opposite.
+  - Since Spark 2.4, when there is a struct field in front of the IN operator before a subquery, the inner query must contain a struct field as well. In previous versions, instead, the fields of the struct were compared to the output of the inner query. E.g. if `a` is a `struct(a string, b int)`, in Spark 2.4 `a in (select (1 as a, 'a' as b) from range(1))` is a valid query, while `a in (select 1, 'a' from range(1))` is not. In previous version it was the opposite.
```
Comment (srowen) [inline docs/sql-migration-guide.md:336]: Nit, for if we need to fix anything else: "E.g." probably should not begin a sentence. "For example," is better

File: docs/sql-migration-guide.md [hunk 2]
```diff
@@ -532,11 +532,11 @@ license: |
...
 
-    - The rules to determine the result type of an arithmetic operation have been updated. In particular, if the precision / scale needed are out of the range of available values, the scale is reduced up to 6, in order to prevent the truncation of the integer part of the decimals. All the arithmetic operations are affected by the change, ie. addition (`+`), subtraction (`-`), multiplication (`*`), division (`/`), remainder (`%`) and positive module (`pmod`).
+    - The rules to determine the result type of an arithmetic operation have been updated. In particular, if the precision / scale needed are out of the range of available values, the scale is reduced up to 6, in order to prevent the truncation of the integer part of the decimals. All the arithmetic operations are affected by the change, i.e. addition (`+`), subtraction (`-`), multiplication (`*`), division (`/`), remainder (`%`) and positive module (`pmod`).
```
Comment (srowen) [inline docs/sql-migration-guide.md:535]: Also let's fix "positive module" to "positive modulus"

**Review by srowen — APPROVED** [review]
Looks fine; I had a few very minor additional changes to suggest

**Comment by kiszk** [comment]
Could you please address the following typos, too?

`docs/configuration.md`
excluded -> excluded

`docs/graphx-programming-guide.md`
an \`VertexRDD -> a \`VertexRDD (2 places)

`docs/mllib-clustering.md`
an \`JavaRDD` -> a \`JavaRDD`

`docs/mllib-data-types.md`
an CoordinateMatrix -> a CoordinateMatrix

`docs/running-on-kubernetes.md` and `docs/running-on-yarn.md`
propogate -> propagate (2 places)

`docs/sql-ref-syntax-qry-select-groupby.md`
column -> column (3 places)

`docs/sql-ref-syntax-qry-select-lateral-view.md`
tabelName -> tableName (3 places)

**Comment by jsoref** [comment]
@kiszk: done (not quite sure how I missed `tabel`)

**Comment by srowen** [comment]
Jenkins test this please

**Comment by kiszk** [comment]
Thanks

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master.

**Pull request closed without merging**